purchases/src/main/kotlin/com/revenuecat/purchases/Purchase.kt

```diff
@@ -0,0 +1,77 @@
+package com.revenuecat.purchases
+
+import android.app.Activity
+import com.revenuecat.purchases.models.GoogleProrationMode
+import com.revenuecat.purchases.models.PurchasingData
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.SubscriptionOption
+
+// TODO BC5 naming
+open class Purchase(builder: Builder) {
+
+    val isPersonalizedPrice: Boolean
+    val oldProductId: String?
+    val googleProrationMode: GoogleProrationMode
+    internal val purchasingData: PurchasingData
+    internal val activity: Activity
+
+    init {
+        this.isPersonalizedPrice = builder.isPersonalizedPrice
+        this.oldProductId = builder.oldProductId
+        this.googleProrationMode = builder.googleProrationMode
+        this.purchasingData = builder.purchasingData
+        this.activity = builder.activity
+    }
+
+    open class Builder private constructor(
+        @get:JvmSynthetic internal val purchasingData: PurchasingData,
+        @get:JvmSynthetic internal val activity: Activity,
+    ) {
+        // TODO jvmsynthetic?
+        // TODO see if there's a way to clean up the purchasingData logic
```

Comment:
was having trouble creating a constructor body for the secondary constructors for some reason, didn't spend a lot of time on it. but i think the purchasingData logic will be moved out to the storeproduct soon anyway, so this should be cleaner

Reply:
I don't think it's too bad anyways